#include "QtConfig.h"

#include "config.h"

int getIndex(vector<pair<string, string>> vec, string value)
{
	for (int i = 0; i < vec.size(); i++)
	{
		pair<string, string> pair = vec[i];
		if (pair.first == value || pair.second == value)
			return i;
	}
	return 0;
}

QtConfig::QtConfig(QWidget *parent)
	: QMainWindow(parent)
{
	ui.setupUi(this);

	Qt::WindowFlags flags = this->windowFlags();
	this->setWindowFlags(flags &~ Qt::WindowMinMaxButtonsHint);
	this->setMinimumSize(this->size());
	this->setMaximumSize(this->size());

	updateInputs();
}

QtConfig::~QtConfig()
{
}

void QtConfig::updateInputs()
{
	ui.isRecordingCheckbox->setChecked(config->isRecording);
	ui.descriptionLine->setText(QString::fromStdString(config->description));
	ui.isTalkerCheckbox->setChecked(config->isTalker);
	ui.isChannelCommanderCheckbox->setChecked(config->isChannelCommander);
	ui.additionalCommandLine->setText(QString::fromStdString(config->additionalCommand));
}

void QtConfig::saveToConfig()
{
	config->isRecording = ui.isRecordingCheckbox->isChecked();
	config->description = ui.descriptionLine->text().toStdString();
	config->isTalker = ui.isTalkerCheckbox->isChecked();
	config->isChannelCommander = ui.isChannelCommanderCheckbox->isChecked();
	config->additionalCommand = ui.additionalCommandLine->text().toStdString();

	config->writeConfig();
}